apps/examples/testcase/ta_tc/audio/utc/utc_audio_main.c

```diff
@@ -0,0 +1,848 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file tc_audio_main.c
+/// @brief Main Function for Filesystem TestCase Example
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <pthread.h>
+#include <dirent.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <tinyara/audio/audio.h>
+#include <tinyalsa/tinyalsa.h>
+#include <tinyara/fs/ioctl.h>
+#include <apps/shell/tash.h>
+#include "tc_common.h"
+
+/****************************************************************************
+ * Definitions
+ ****************************************************************************/
+#define AUDIO_TEST_FILE "/mnt/pcm"
+
+#define AUDIO_DEFAULT_CHANNELS 2
+
+#define AUDIO_DEFAULT_RATE 16000
+
+#define AUDIO_DEFAULT_FORMAT PCM_FORMAT_S16_LE
+
+#define AUDIO_RECORD_DURATION 3 //3sec
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+
+extern sem_t tc_sem;
+
+extern int working_tc;
+
+struct pcm *g_pcm;
+
+unsigned int g_byte_size;
+
+char *g_record_buffer;
+
+
+/**
+* @testcase         audio_pcm_open_p
+* @brief            open and initlaize pcm structure
+* @scenario         Open pcm and check its validation
+* @apicovered       pcm_open
+* @precondition     NA
+* @postcondition    NA
+*/
+static void utc_audio_pcm_open_tc_p(void)
+{
+	g_pcm = pcm_open(0, 0, PCM_IN, NULL);
+	TC_ASSERT_GT("pcm_open", pcm_get_file_descriptor(g_pcm), 0)
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_open_n
+* @brief            open and initlaize pcm structure with invalid values
+* @scenario         Open pcm with invalid device & card value and check its validation
+* @apicovered       pcm_open
+* @precondition     NA
+* @postcondition    NA
+*/
+static void utc_audio_pcm_open_tc_n(void)
+{
+	struct pcm *pcm;
+	pcm = pcm_open(999, 999, PCM_IN, NULL);
+	TC_ASSERT_LEQ("pcm_open", pcm_get_file_descriptor(pcm), 0);
+	pcm_close(pcm);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         audio_pcm_is_ready_p
+* @brief            open and check pcm is ready or not
+* @scenario         Open pcm and check its state
+* @apicovered       pcm_is_ready
+* @precondition     pcm_open should be called
+* @postcondition    NA
+*/
+static void utc_audio_pcm_is_ready_tc_p(void)
+{
+	TC_ASSERT("pcm_is_ready", pcm_is_ready(g_pcm));
```

Comment:
TC_ASSERT will be deleted. Lets change it to EQ or NEQ

Reply:
ok will do that I didn't get any information about it.